app/Kconfig

```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2021 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+#
+# This file is the application Kconfig entry point. All application Kconfig
+# options can be defined here or included via other application Kconfig files.
+# You can browse these options using the west targets menuconfig (terminal) or
+# guiconfig (GUI).
+
+menu "Zephyr"
```

Comment:
> Using "Zephyr" as a menu name is confusing. Please update to indicate which repository/module this is for (e.g. "Example Application" or "Zephyr Project's Example Application).

Why? I think this is good as-is. If you open it in menuconfig you'll see that it's a menu for the options that are built into zephyr.

Reply:
This menu encapsulates Zephyr's Kconfig, so I think "Zephyr" name is appropriate.

Reply:
Consider what happens when there are multiple "application" modules active at once.  With this rationale they will all report "Zephyr", which is more confusing than helpful so it does not scale well.

Reply:
@tejlmand When I integrate multiple non-zephyrized open source projects with my proprietary content, I do the following **for each repository**:
1. fork repository
2. create a long-term integration branch for my extensions
3. add a manifest importing the Zephyr version I will integrate against
3. add files and empty test cases so it is picked up by my Zephyr build and CI systems
4. add configuration settings, drivers, subsystems, and samples as "glue code" and for verification of integration

When used stand-alone, each of these are an application repository.  When pulled together in my top-level "platform" (manifest+module) repository, all the Kconfig of all these application modules are built together, with drivers and subsystems of each being integrated in one executable.  Currently I'm using about a handfull of these intermediate manifest+module repositories.  I do not fold these together into a single repo because they are from different open source projects that I need to track.

Reply:
@tejlmand Please remember also that the [T3: Forest topology](https://docs.zephyrproject.org/latest/guides/west/workspaces.html?highlight=forest#t3-forest-topology) illustrates multiple application directories at the top level, which naturally leads to multiple application modules.  :-)

Reply:
@gregshue each of those would get their own version of this file. You cannot configure more than one application at a time in a single menuconfig instance.